removed ESH references and added governance infos (#885)
openhab/openhab-docs: developers/contributing/contributing.md

* [openHAB 1 Add-ons](https://github.com/openhab/openhab1-addons): Legacy add-ons that were developed for openHAB 1. Most of them are working smoothly on the openHAB 2 runtime and thus they are made available for backward compatibility reasons. They are not suggested for new users, though.
* [openHAB Distro](https://github.com/openhab/openhab-distro): This repo contains all parts that are required for assembling the binary distribution of openHAB.

Besides those "main" repositories, there are a dozen further ones, which are not listed here in detail - just [check out the list](https://github.com/openhab) and you will find separate repos for openHAB Cloud, the Alexa skill, the Google Action, the Android and iOS apps, the Zigbee and ZWave bindings, HABPanel and HABot and many more.
If you want to get involved with any of these, best check their respective README and CONTRIBUTING files in the root of the repos.

## Contribution Guidelines

### Pull Requests are Always Welcome

We are always thrilled to receive pull requests, and do our best to process them as fast as possible.
Not sure if that typo is worth a pull request? Do it! We will appreciate it.

> If your pull request is not accepted on the first try, don't bediscouraged!

Reviewer: ```suggestion
If your pull request is not accepted on the first try, don't be discouraged!
```

> openHAB maintainers use the [Github review feature](https://help.github.com/articles/about-pull-request-reviews/) to indicate acceptance.

Reviewer: ```suggestion
openHAB maintainers use the [GitHub review feature](https://help.github.com/articles/about-pull-request-reviews/) to indicate acceptance.
```

* Step 1: learn the component inside out
* Step 2: make yourself useful by contributing code, bugfixes, support etc.

> * Step 3: volunteer [in the community] (https://community.openhab.org/c/development/misc) or in the appropriate Github project and have a [maintainer](governance.html#maintainers) [nominate you](governance.html#nominations).

Reviewer: ```suggestion
* Step 3: volunteer [in the community] (https://community.openhab.org/c/development/misc) or in the appropriate GitHub project and have a [maintainer](governance.html#maintainers) [nominate you](governance.html#nominations).
```
